### PR TITLE
Sils/down

### PR DESCRIPTION
**Pull request by sils1297**
CC @Adrianzatreanu

### CODE REVIEW

File: coalib/bears/Bear.py [hunk 1]
```diff
@@ -217,3 +228,15 @@ def get_config_dir(self):
...
+    @classproperty
+    def data_dir(cls):
+        """
+        Returns a directory that may be used by the bear to store stuff. Every
+        bear has an own directory dependent on his name.
+        """
+        data_dir = abspath(join(user_data_dir('coala-bears'), cls.name))
+
+        if not isdir(data_dir):  # pragma: no cover
+            makedirs(data_dir)
```
Comment (Makman2) [inline coalib/bears/Bear.py]: I think you can use `exist_ok=True` as a parameter. Then you don't need the if :) (and also no pragma no cover)

**Comment by Makman2** [comment]
unack c8381dd

File: coalib/bears/Bear.py [hunk 2]
```diff
@@ -229,6 +231,44 @@ def get_config_dir(self):
...
+
+        If we download it again, it'll be much faster as no download occurs:
+
+        >>> newfile = bear.download_cached_file("http://gitmate.com/", "a_file")
+        >>> newfile == file
+        True
+
+        :param url: The URL to download the file from.
+        :param filename: The filename it should get, e.g. "test.txt".
+        :return: A full path to the file ready for you to use!
```
Comment (Makman2) [inline coalib/bears/Bear.py]: Alignment :)

**Comment by Makman2** [comment]
ack faa32a6

**Comment by Makman2** [comment]
I've thought about moving the `setup_dependencies` stuff into `check_prerequisites`, but from users perspective this is untuitive and bad, as the user could forget to invoke the old check-prerequisites again^^

**Comment by sils1297** [comment]
it's also two different things

**Comment by sils1297** [comment]
reack adaf631

File: coalib/bears/Bear.py [hunk 3]
```diff
@@ -217,3 +228,14 @@ def get_config_dir(self):
...
+
+    @classproperty
+    def data_dir(cls):
+        """
+        Returns a directory that may be used by the bear to store stuff. Every
+        bear has an own directory dependent on his name.
+        """
+        data_dir = abspath(join(user_data_dir('coala-bears'), cls.name))
+
+        makedirs(data_dir)
```
Comment (Makman2) [inline coalib/bears/Bear.py]: wait you need `exist_ok=True` as parameter, otherwise it raises an error :O

Comment (Makman2) [inline coalib/bears/Bear.py]: (if dir already exists)

**Comment by Makman2** [comment]
unack 7625f68

**Comment by Makman2** [comment]
ack 90b8d39

**Comment by sils1297** [comment]
108e01b Bear: Add setup_dependencies hook
38588e9 Bear: Add download_cached_file
reack

**Comment by Makman2** [comment]
ack 
	1b15d1d

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**